libs/lib-client/header/src/lib/header.component.html

```diff
@@ -0,0 +1,37 @@
+<div class="header">
+  <i
+    class="flogo-icon flogo-icon-back header-back qa-icon-back"
+    [class.--hover]="isHoveredOnBack"
+    (click)="goBack.emit()"
+    (mouseover)="onMouseOverBackControl()"
+    (mouseout)="onMouseOutBackControl()"
+  ></i>
+  <ng-content select="[header-resource-logo]"></ng-content>
+  <div class="header-content">
+    <div class="resource-app-details">
+      <div
+        class="resource-name qa-resource-name"
+        [fgContentEditable]="name"
+        (fgContentEditableChange)="changeName.emit($event)"
+      ></div>
+      <span>in</span>
+      <span
+        class="resource-app-name qa-app-name"
+        (click)="goBack.emit()"
+        [class.--hover]="isHoveredOnBack"
+        (mouseover)="onMouseOverBackControl()"
+        (mouseout)="onMouseOutBackControl()"
+        >{{ appName }}</span
+      >
+    </div>
+    <p
+      class="resource-description qa-resource-description"
+      [fgContentEditable]="description"
+      (fgContentEditableChange)="changeDescription.emit($event)"
+      [allowNewLines]="true"
+      placeholder="{{ 'CANVAS:ADD-FLOW-DESCRIPTION' | translate }}"
+    ></p>
+  </div>
+  <ng-content select="[header-right-block]"></ng-content>
```

Comment:
Do you think using a selector like `[header-role="side-area"]` would look better?

And then usage:
```html
<flogo-designer-header>
    <div header-role="resource-icon"></div>
	<div header-role="side-area"></div>
    <div header-role="bottom-area"></div>
<flogo-designer-header/>
```